File: Coursework/Level_2/financial_record.py
# SACCO Member Financial Record in Python

def savings_calculator():
    mbr_name = input("Enter your name: ")
    mbr_id = input("Enter your Member ID no.: ")

    first_month_cntb = int(input("Enter your first month contribution: "))
    second_month_cntb = int(input("Enter your second month contribution: "))
    thrid_month_cntb = int(input("Enter your thrid month contribution: "))
    fourth_month_cntb = int(input("Enter your fourth month contribution: "))
    fifth_month_cntb = int(input("Enter your fifth month contribution: "))
    sixth_month_cntb = int(input("Enter your sixth month contribution: "))

    total_savings = first_month_cntb + second_month_cntb + thrid_month_cntb + fourth_month_cntb + fifth_month_cntb + sixth_month_cntb

    print("\n------------------------------")
    print(f"Member name: {mbr_name}")
    print(f"Member ID: {mbr_id}")
    print(f"Total savings for the last 6 months: {total_savings}")
    print("------------------------------\n")


if __name__ == "__main__":
    savings_calculator()

File: Coursework/Level_2/FinancialRecord.java
import java.util.Scanner;

public class FinancialRecord {
    private String memberName;
    private String memberID;
    private double[] monthlyContributions;
    private double totalSavings;
    
    // Constructor
    public FinancialRecord() {
        monthlyContributions = new double[6];
        totalSavings = 0;
    }
    
    // Method to input member details
    public void inputMemberDetails() {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter Member Name: ");
        memberName = scanner.nextLine();
        
        System.out.print("Enter Member ID: ");
        memberID = scanner.nextLine();
    }
    
    // Method to input 6 months of contributions
    public void inputMonthlyContributions() {
        Scanner scanner = new Scanner(System.in);
        
        System.out.println("\nEnter monthly contributions for 6 months:");
        for (int i = 0; i < 6; i++) {
            System.out.print("Month " + (i + 1) + " contribution: ");
            monthlyContributions[i] = scanner.nextDouble();
        }
    }
    
    // Method to calculate total savings
    public void calculateTotalSavings() {
        totalSavings = 0;
        for (int i = 0; i < 6; i++) {
            totalSavings += monthlyContributions[i];
        }
    }
    
    // Method to display member record
    public void displayRecord() {
        System.out.println("\n========== FINANCIAL RECORD ==========");
        System.out.println("Member Name: " + memberName);
        System.out.println("Member ID: " + memberID);
        System.out.println("Total Savings: " + totalSavings);
        System.out.println("=====================================\n");
    }
    
    // Main method
    public static void main(String[] args) {
        FinancialRecord record = new FinancialRecord();
        
        record.inputMemberDetails();
        
        record.inputMonthlyContributions();
        
        record.calculateTotalSavings();
        
        record.displayRecord();
    }
}
